12/umbraco-cms/tutorials/getting-started-with-entity-framework-core.md

```diff
@@ -0,0 +1,226 @@
+
+    public required string Name { get; set; }
+
+    public required string Email { get; set; }
+
+    public required string Website { get; set; }
+
+    public string Message { get; set; } = string.Empty;
+}
+```
+
+Now that we have the model, lets create a `DbContext` class so we can interact with the database, we'll call this `BlogContext`.
+
+### BlogContext class
+```
+using Microsoft.EntityFrameworkCore;
+
+namespace Umbraco.Demo;
+
+public class BlogContext : DbContext
+{
+    public BlogContext(DbContextOptions<BlogContext> options)
+        : base(options)
+    {
+    }
+
+    public required DbSet<BlogComment> BlogComments { get; set; }
+
+    protected override void OnModelCreating(ModelBuilder modelBuilder) =>
+        modelBuilder.Entity<BlogComment>(entity =>
+        {
+            entity.ToTable("blogComment");
+            entity.HasKey(e => e.Id);
+            entity.Property(e => e.Id).HasColumnName("id");
+            entity.Property(e => e.BlogPostUmbracoId).HasColumnName("blogPostUmbracoKey");
+            entity.Property(e => e.Message).HasColumnName("message");
+            entity.Property(e => e.Website).HasColumnName("website");
+            entity.Property(e => e.Email).HasColumnName("email");
+        });
+}
+```
+
+We need to register this DbContext to use it, in Umbraco because we need to register a few other things to have Scopes working, we have a helpful extension method you use:
+`services.AddUmbracoEFCoreContext<BlogContext>("{YOUR CONNECTIONSTRING HERE}", "{YOUR PROVIDER NAME HERE}");`
+
+Now that we can access the database via the `BlogContext`, we need to first migrate the database to add our tables.
+With EFCore, you can autogenerate the migrations via the console, so lets open up a terminal and do it!
+
+Open your terminal and navigate to your project folder.
+If you do not have the EFCore CLI tool installed, run this command to install it: `dotnet tool install --global dotnet-ef`
+We can then generate the migration by running this command: `dotnet ef migrations add InitialCreate --context BlogContext`
+
+{% hint style="warning" %}
+This example named the migration `InitialCreate` but you can choose whatever name you like!
+Our DbContext class was named `BlogContext`, but if you have renamed yours, remember to also change that in the command!
+{% endhint %}
+
+You should now have a `Migrations` folder in your project, containing the `InitialCreate` migration (or whatever name you chose).
+
+{% hint style="warning" %}
+This might be confusing at first, as when working with EFCore you would usually just inject your `Context` class.
+And while you can still do that, it is however not the recommended approach in Umbraco.
+In Umbraco we use a concept called `Scope` which is our implementation of the `Unit of work` pattern.
+This ensures that we start a transaction when using the database, and roll it back if the scope is not completed (for example when expections are thrown).
```

Comment:
```suggestion
This ensures that we start a transaction when using the database. If the scope is not completed (for example when exceptions are thrown) it will roll it back.
```